### PR TITLE
[CAS/ActionCache] Add asynchronous versions for the `ActionCache` APIs

### PR DESCRIPTION
**Pull request by akyrtzi**

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: llvm/unittests/CAS/ActionCacheTest.cpp [hunk 1]
```diff
@@ -106,3 +106,45 @@ TEST(OnDiskActionCache, ActionCacheResultInvalid) {
   // Write a different value will cause error.
   ASSERT_THAT_ERROR(Cache2->put(*ID3, *ID3), Failed());
 }
+
+TEST_P(CASTest, ActionCacheAsync) {
+  std::shared_ptr<ObjectStore> CAS = createObjectStore();
+  std::unique_ptr<ActionCache> Cache = createActionCache();
+
+  {
+    Optional<ObjectProxy> ID;
```
Comment (cachemeifyoucan) [inline llvm/unittests/CAS/ActionCacheTest.cpp]: llvm::Optional mixed with std::optional?

**Review by cachemeifyoucan — APPROVED** [review]
LGTM with a small comment.

**Review by benlangmuir — APPROVED** [review]

File: llvm/unittests/CAS/ActionCacheTest.cpp (hunk 1 above)
Comment (akyrtzi) [reply llvm/unittests/CAS/ActionCacheTest.cpp]: Good catch! Fixed.

**Pull request merged**